backend/de.metas.business/src/main/java/de/metas/costing/CurrentCost.java

```diff
@@ -156,7 +156,17 @@ public void addWeightedAverage(
 		assertCostCurrency(amt);
 		Check.assume(qty.signum() != 0, "qty not zero");
 
-		final CostAmount currentAmt = costPrice.getOwnCostPrice().multiply(currentQty);
+		final CostAmount currentAmt;
+
+		if (currentQty.isZero())
+		{
+			currentAmt = costPrice.getOwnCostPrice();
```

Comment:
not sure this is ok....

Reply:
In case a new inbound request (from M_InventoryLine or M_Match_PO) is added, the currentCost calculation must take into account the already existing price even if the existing qty is 0.

Reply:
example: 
* An old inventory with qty 0 and cost 100 already exists
* Now we purchase 10 pieces with the price of 80 / piece, 
* MatchPO is created
* => the currentCost should be (100 + 800)/10 = 90